### ISSUE TITLE
Incorrect VM Snapshot Id in API response

### ISSUE THREAD

**Issue by satishglondhe** [apache/cloudstack#6444]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
API
~~~

##### CLOUDSTACK VERSION

~~~
CloudStack 4.16.1.0
~~~

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

While creating VM Snapshot using APIs working fine but in the response it returns int id instead of UUID which is required to do the further snapshot operations.

##### STEPS TO REPRODUCE

Create VM Snapshot using API *createVMSnapshot*
apiref: https://cloudstack.apache.org/api/apidocs-4.16/apis/createVMSnapshot.html

##### EXPECTED RESULTS

~~~
It should return UUID of the snapshot

~~~

##### ACTUAL RESULTS

The createVMSnapshot job returning the int id instead of snapshot UUID
Current API response is
~~~
{"createvmsnapshotresponse":{"id":"39","jobid":"c89c1aa2-c34c-478a-810e-b221c2a1076c"}}
~~~

**Comment by nvazquez** [comment]
Hi @satishglondhe - the create VM snapshot operation is an asynchronous job, it is retrieving the jobid on the initial response while the job is in progress. The UUID (and other relevant fields) are set once the operation is completed and successful. Have you tried using cloudmonkey (https://github.com/apache/cloudstack-cloudmonkey)? - it handles these cases and queries the in-progress jobs until its completion, retrieving the full response once the operation is completed

**Comment by nvazquez** [comment]
Fixed by #6449
